python/pyspark/errors/error-conditions.json

```diff
@@ -189,11 +189,21 @@
       "Remote client cannot create a SparkContext. Create SparkSession instead."
     ]
   },
+  "DATA_SOURCE_EXTRANEOUS_FILTERS": {
+    "message": [
+      "<type>.pushFilters() returned <extraneous> that are not part of the input <input>. Make sure that the returned filters are a subset of the input filters."
+    ]
+  },
   "DATA_SOURCE_INVALID_RETURN_TYPE": {
     "message": [
       "Unsupported return type ('<type>') from Python data source '<name>'. Expected types: <supported_types>."
     ]
   },
+  "DATA_SOURCE_PUSHDOWN_DISABLED": {
+    "message": [
+      "<type> implements pushFilters() but filter pushdown is disabled because '<conf>' is False. Run 'spark.conf.set(\"<conf>\", True)' to enable filter pushdown."
```

Comment:
`spark.conf.set(\"<conf>\", True)` is python syntax. We can just ask users to set the conf to true, without mentioning the concrete API.